@xen-orchestra/web/src/main.ts

```diff
@@ -7,19 +8,35 @@ import App from './App.vue'
 import '@xen-orchestra/web-core/assets/css/base.pcss'
 import 'vue-virtual-scroller/dist/vue-virtual-scroller.css'
 
-const app = createApp(App)
+async function init() {
+  // TODO: remove when non admin users are handled in the REST API
+  if (import.meta.env.PROD) {
+    // Expect an error for no admin users.
+    // redirect to XO5 in that case
+    const { statusCode } = await useFetch('/rest/v0/users/me')
+    if (statusCode.value === 403) {
+      // voluntary not using `useXoRoutes` we need info BEFORE mounting anything from XO6 (UI,store, ...)
+      const { data } = await useFetch<{ xo5: string; xo6: string }>('/rest/v0/gui-routes').json()
+      window.location.pathname = data.value!.xo5
```

Comment:
I don't think using '!' is really good ? Wouldn't it be possible to check if `data.value` exists?

```suggestion
     if (data.value) {
           window.location.pathname = data.value.xo5
        }
```

Reply:
In theory, if `/rest/users/me` returns a 403 status code, then `data.value` cannot be undefined.

I'll add a check for clearer code.